Documentation/Administration/Deployment/Automated/Index.rst

```diff
@@ -0,0 +1,195 @@
+
+..  _ci-cd-common-stages:
+
+Common CI/CD Stages in TYPO3 Projects
+=====================================
+
+..  _ci-cd-code-quality-checks:
+
+Code Quality Checks
+-------------------
+
+*   **PHP Linting** (for example, :bash:`php -l`)
+*   `Code style tests and fixing <https://docs.typo3.org/permalink/t3coreapi:testing-projects-configuration-cs>`_
+*   **Static Analysis** (for example `PHPstan <https://docs.typo3.org/permalink/t3coreapi:testing-projects-configuration-phpstan>`_ or Psalm)
+*   Linting of other used formats like `TypoScript <https://github.com/martin-helmich/typo3-typoscript-lint>`_, XML, and YAML.
+
+..  _ci-cd-unit-functional-testing:
+
+Unit and Functional Testing
+---------------------------
+
+*   `Unit tests <https://docs.typo3.org/permalink/t3coreapi:testing-writing-unit>`_
+*   `Functional tests <https://docs.typo3.org/permalink/t3coreapi:testing-writing-functional>`_
+*   `Acceptance tests <https://docs.typo3.org/permalink/t3coreapi:testing-writing-acceptance>`_
+
+..  _ci-cd-building-artifacts:
+
+Building Artifacts
+------------------
+
+*   Installing required extensions and other PHP packages via Composer
+*   Compile frontend assets (e.g., SCSS, JavaScript) using tools like
+    Webpack, Gulp or Vite.
+
+..  _ci-cd-deployment:
+
+Deployment
+----------
+
+*   **File Synchronization**: Deploy code and assets using tools like `Rsync <https://docs.typo3.org/permalink/t3coreapi:deployment-rsync>`_,
+    `Deployer <https://docs.typo3.org/permalink/t3coreapi:deployment-deployer>`_,
+    or Git-based workflows.
+*   **Database Migrations**: Run database migrations using TYPO3’s
+    `vendor/bin/typo3 extension:setup` or
+    `vendor/bin/typo3 database:updateschema` if :composer:`helhum/typo3-console`
+    is installed.
+*   **Cache Clearing**: Clear TYPO3 caches
+    (`vendor/bin/typo3 cache:flush`).
+
+..  _ci-cd-environment-configuration:
+
+Environment configuration
+--------------------------
+
+Manage environment-specific settings using
+`.env / dotenv files <https://docs.typo3.org/permalink/t3coreapi:environment-dotenv>`_
+or `Plain PHP configuration files <https://docs.typo3.org/permalink/t3coreapi:environment-phpconfig>`_.
+
+..  _ci-cd-typical-tools:
+
+Typical CI/CD Tools Used
+------------------------
+
+CI/CD Platforms
+    `GitHub Actions <https://docs.typo3.org/permalink/t3coreapi:ci-cd-github>`_,
+    `GitLab CI <https://docs.typo3.org/permalink/t3coreapi:ci-cd-gitlab>`_,
+    `Jenkins <https://www.jenkins.io/>`_, or `CircleCI <https://circleci.com/>`_
+
+Code Quality
+    `PHP-CS-Fixer <https://docs.typo3.org/permalink/t3coreapi:testing-projects-configuration-cs>`_,
+    `PHPstan <https://docs.typo3.org/permalink/t3coreapi:testing-projects-configuration-phpstan>`_,
+    and `TypoScript-Lint <https://github.com/martin-helmich/typo3-typoscript-lint>`_
+
+Testing
+    `PHPUnit <https://phpunit.de/>`_ and `TYPO3 Functional Testing Framework <https://docs.typo3.org/permalink/t3coreapi:testing-writing-unit>`_
```

Comment:
The 2nd link says "functional" but links to "unit" …

Better make both generic to mention the TF itself.

Reply:
We dont have an extra chapter on the TF, we have one on unit tests and one on funcitonal tests, both using the TF. But removed the funcitonal